apps/system/lockscreen/js/charging.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+(function(exports) {
+
+/**
+ * Creates an object used for refreshing the charging string UI element.
+ * @class  ChargingStatus
+ */
+function ChargingStatus() {
+  this.refreshCB = null;
+}
+
+ChargingStatus.prototype.start = function cs_start() {
+  this.refresh();
+
+  if (!this.refreshCB) {
+    this.refreshCB = this.refresh.bind(this);
+
+    navigator.battery.addEventListener('chargingchange',
+      this.refreshCB);
+    navigator.battery.addEventListener('levelchange',
+      this.refreshCB);
+    navigator.battery.addEventListener('chargingtimechange',
+      this.refreshCB);
+  }
+};
+
+ChargingStatus.prototype.stop = function cs_stop() {
+  if (this.refreshCB) {
+    navigator.battery.removeEventListener('chargingchange',
+      this.refreshCB);
+    navigator.battery.removeEventListener('levelchange',
+      this.refreshCB);
+    navigator.battery.removeEventListener('chargingtimechange',
+      this.refreshCB);
+    this.refreshCB = null;
+  }
+};
+
+ChargingStatus.prototype.refresh = function cs_refresh() {
+  var lockscreenCharging = document.getElementById('lockscreen-charging'); 
```

Comment:
I prefer to have a structure to contain all elements as:

    this.elements = {};  // In the constructor
    this.elements.charging = document.getElementById...

Since I had seen such notorious code in the old `lockscreen.js`...

    this.passCode = document.getElementById(...)
    this.passcodeCode = '';  // The passcode, in string

And no one would know the difference between names like this after 3 weeks. So I think to have some hints for this is important. Moreover, I think since you're implement a comonent under LockScreen, to have the element without 'lockscreen' previs is OK. It's also more clear to have `this.elements.charging` than `this.lockscreenCharging`, which looks like that you would have other not-lockscreen charging elements in this component.